src/app/layout.tsx

```diff
@@ -38,7 +38,7 @@ export default function RootLayout({ children }: { children: ReactNode }) {
 	const path = usePathname();
 	const storedToken = getToken();
 	const [egoJwt, setEgoJwt] = useState(storedToken || '');
-	const initLoginState = path === '/logged-in' && !storedToken ? true : false;
+	const initLoginState = path === '/logged-in' && !(storedToken || egoJwt) ? true : false;
```

Comment:
From the conditional logic here, I find quite hard to figure out what `initLoginState` as a boolean is supposed to mean. e.g. is it short for `wasUserLoggedInWhenComponentLoaded`? if so, why does having a token/jwt lead to `false`? the conditions read to me like a `wasUserNotLoggedInWhenComponentLoaded`, but I'm probably missing something here...

Reply:
Fair, I'll update. I'm using this to manage the NavBar Login button / Profile badge state: 
- if you're on the homepage with no token, you should see Login
- if you're on the Logging In page you should see nothing
- if you have a token, you should see the Profile badge

Reply:
All that from that little bit of logic? wow. 
💪 heavy lifting!

In that case, it almost seems like all this logic should be abstracted into the auth provider, and passed from there to the relevant children (as individual variables, like `shouldShowBadge`, `shouldRedirectToLogin`, etc.). 
Unless you foresee this being relevant at the layout base in the future?